### PR TITLE
variable: add global, session variable cte_max_recursion_depth

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/24108

### Release note 

- add global | session variable cte_max_recursion_depth

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -826,6 +826,10 @@ type SessionVars struct {
 
 	// EnableDynamicPrivileges indicates whether to permit experimental support for MySQL 8.0 compatible dynamic privileges.
 	EnableDynamicPrivileges bool
+
+	// CteMaxRecursionDepth indicates The common table expression (CTE) maximum recursion depth.
+	// see https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_cte_max_recursion_depth
+	CteMaxRecursionDepth int
```
Comment (xhebox) [inline sessionctx/variable/session.go:832]: Will it be better to name it `CTEMaxRecursionDepth`?

**Review by xhebox — COMMENTED** [review]
Rest LGTM

File: sessionctx/variable/session.go (hunk 1 above)
Comment (xiongjiwei) [reply sessionctx/variable/session.go]: PTAL

**Review by xhebox — APPROVED** [review]

**Pull request merged**